### PR TITLE
postgresqlPackages.pgvecto-rs: init at 0.2.1

### LINKED ISSUE #274509: Package request: pgvecto.rs
**Project description**

Scalable Vector database plugin for Postgres, written in Rust, specifically designed for LLM

Immich 1.91 requires this extension for it's PostreSQL database.

**Metadata**

* homepage URL: https://github.com/tensorchord/pgvecto.rs
* source URL: https://github.com/tensorchord/pgvecto.rs
* license: asl20
* platforms: unix, linux, darwin, ...

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by diogotcorreia**
## Description of changes

Description: Scalable Vector Search in Postgres. Revolutionize Vector Search, not Database. 
https://github.com/tensorchord/pgvecto.rs

Used for [Immich](https://github.com/immich-app/immich).

Closes #274509

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

~~I have not tested this with Immich before, but as per https://github.com/NixOS/nixpkgs/issues/274509#issuecomment-1892563516, it seems like this version would not work (because of upstream).~~

~~Keeping this as a draft until Immich moves to a more recent pgvecto.rs version.~~

Edit: see https://github.com/NixOS/nixpkgs/pull/281192#issuecomment-1937355090 (next version of Immich will use pgvecto.rs v0.2.0)

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by diogotcorreia** [comment]
It looks like v0.2.0 has been released and that Immich is moving to that version of pgvecto.rs in the next release: https://github.com/immich-app/immich/pull/6785

I'll be working tomorrow to get this ready in time for the release!

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (Atemu) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:14]: Either way would be acceptable; we don't know how well it'd take newer versions. I'd try to make the original Nixpkgs package work and make it capable of multi-version if push comes to shove but this is fine too.

Either way, add this info as a code comment.

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (Atemu) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:9]: I don't know this section of Nixpkgs very well but, reading the source a bit, it appears you can create your own `rustPlatform` using `makeRustPlatform` and pass it your desired version of `clang`.

i.e.

```Nix
makeRustPlatform {
  clang = clang_16;
  inherit rustc cargo;
}
```

https://nixos.org/manual/nixpkgs/stable/#using-rust-nightly-in-a-derivation-with-buildrustpackage

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (Atemu) [inline pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:55]: Not familiar with rust but what exactly does this mean?

**Review by Atemu — COMMENTED** [review]
This already looks quite good. A couple minor things.

I'd also like to know how you test this works as expected.

File: pkgs/servers/sql/postgresql/packages.nix
Comment (Atemu) [inline pkgs/servers/sql/postgresql/packages.nix:48]: @infinisil by-name CI fails here but I don't think it should since this isn't all-packages.nix.

Comment (diogotcorreia) [reply pkgs/servers/sql/postgresql/packages.nix:48]: It only started failing after I edited the PR description last night btw
I'll also take a look to see if it's something I can fix on my side, but it seems to be a problem with the CI at first glance

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (diogotcorreia) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:55]: This means they have at least one dependency that is fetched from a git repo instead of from crates.io.
https://github.com/tensorchord/pgvecto.rs/blob/be5a816810f63cadada90a666cfc70db37c78a33/Cargo.toml#L35

I can try to clarify more in the comment and put there the link above as well. I took a look at the master branch now and that dependency is gone, so it seems this might stop being a problem in the future.

File: pkgs/servers/sql/postgresql/packages.nix
Comment (diogotcorreia) [reply pkgs/servers/sql/postgresql/packages.nix:48]: Seems to have been caused by commit https://github.com/NixOS/nixpkgs/commit/a8a7369c36120acb9ca43376504b6e9babed7261, which was reverted in the meantime, https://github.com/NixOS/nixpkgs/commit/ac637ef21ecc2f0a1a0987fe9ce1d7f0d7f42fca.
I've edited the PR description again to trigger the CI and it passes now.

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (Atemu) [inline pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:20]: Relative paths are not part of a package's interface. We'll have to find a different solution.

Comment (diogotcorreia) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:20]: You're absolutely right. I'll investigate alternative solutions

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (Atemu) [inline pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:87]:
```suggestion
    substituteInPlace ./vectors.control --subst-var-by CARGO_VERSION ${version}
```

**Comment by PhilippWoelfel** [comment]
Can you add the source files from `sql/upgrade` to the output? E.g.,
```
  postInstall = ''
    cp sql/upgrade/* $out/share/postgresql/extension/
  '';
```

**Comment by diogotcorreia** [comment]
@PhilippWoelfel that was also in my todo list as well, yes :)
I was still going to investigate how they do it on their CI, since the normal build process doesn't seem to include them.

I'm not sure if I'm going to be able to finish the remaining problems this with PR this weekend, but I'll do my best :sweat_smile:

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (diogotcorreia) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:21]: There doesn't seem to be a way to override it. This is what I've tried:

```
nix-repl> pkgs.rustPlatform.bindgenHook.override { clang = pkgs.clang_17; }
error:
       … while calling a functor (an attribute set with a '__functor' attribute)

         at «string»:1:1:

            1| pkgs.rustPlatform.bindgenHook.override { clang = pkgs.clang_17; }
             | ^

       … while calling a functor (an attribute set with a '__functor' attribute)

         at /home/dtc/documents/vcs/nixpkgs/lib/customisation.nix:104:43:

          103|       # Re-call the function but with different arguments
          104|       overrideArgs = mirrorArgs (newArgs: makeOverridable f (overrideWith newArgs));
             |                                           ^
          105|       # Change the result of the function call by applying g to it

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: function 'anonymous lambda' called with unexpected argument 'clang'

       at /home/dtc/documents/vcs/nixpkgs/pkgs/build-support/rust/hooks/default.nix:92:32:

           91|
           92|     bindgenHook = callPackage ({}: makeSetupHook {
             |                                ^
           93|       name = "rust-bindgen-hook";

```

```
nix-repl> (pkgs.makeRustPlatform {rustc = pkgs.rustc; cargo = pkgs.cargo; clang = pkgs.clang_17;}).bindgenHook.clang
«derivation /nix/store/j16048gsic45ijdpw82172pfxdz9nw2w-clang-wrapper-16.0.6.drv»
```

For now, I'll setup a similar script just for this package that supports pinning the clang version, but this should be revisited at a later date to allow overriding this.

**Comment by diogotcorreia** [comment]
I think it should be ready now :) :crossed_fingers: 

Changes I've made:
- Check extension version in the tests
- Properly pinned clang 16 by setting an env variable in `preBuild`
- Changed env variable name in patch 0001 to `C_CRATE_EXTRA_CLANG_ARGS` to avoid conflict with `rustPlatform.bindgenHook`
- Replaced use of `sed` with `substituteInPlace` as per the review
- Included the upgrade SQL scripts (thanks @PhilippWoelfel !)
- Updated package description as per upstream

Since I'm force pushing, here's the diff: https://bin.diogotc.com/xuzecylyju.diff

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (Atemu) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix]: This looks like a bug. It can be fixed with this:

```patch
diff --git a/pkgs/development/compilers/rust/make-rust-platform.nix b/pkgs/development/compilers/rust/make-rust-platform.nix
index e22cb6d594af..6ed724aae821 100644
--- a/pkgs/development/compilers/rust/make-rust-platform.nix
+++ b/pkgs/development/compilers/rust/make-rust-platform.nix
@@ -1,4 +1,4 @@
-{ lib, buildPackages, callPackage, cargo-auditable, stdenv, runCommand }@prev:
+{ lib, buildPackages, callPackage, callPackages, cargo-auditable, stdenv, runCommand }@prev:
 
 { rustc
 , cargo
@@ -34,7 +34,7 @@ rec {
   };
 
   # Hooks
-  inherit (callPackage ../../../build-support/rust/hooks {
+  inherit (callPackages ../../../build-support/rust/hooks {
     inherit stdenv cargo rustc;
   }) cargoBuildHook cargoCheckHook cargoInstallHook cargoNextestHook cargoSetupHook maturinBuildHook bindgenHook;
 }
```

Comment (diogotcorreia) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix]: I'll investigate that later then, thanks :)
Should this be made in another PR, or does it make sense to include it in this one?

**Comment by Atemu** [comment]
Re: force pushing obfuscating github logs, you could just push them as simple commits and squash/fixup later. If you had to rebase, force-push the rebase of the original commits first and then the additional commits and/or squash/fixup.

That way changes remain traceable.

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (diogotcorreia) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix]: Great, it seems to work fine in the repl. I've tested it by overriding clang to 17 and having it fail the build.
I'm just finishing building the package to make sure everything works fine, and then I'll push.
This will likely trigger a lot of package rebuilds in the CI, but we'll see :upside_down_face:

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (Atemu) [inline pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:22]: I'd prefer a full `makeRustPlatform` call.

Comment (diogotcorreia) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:22]: I tried that, but it seems to not be overriding as expected.
```
nix-repl> (pkgs.makeRustPlatform {rustc = pkgs.rustc; cargo = pkgs.cargo; clang = pkgs.clang_17;}).bindgenHook.clang
«derivation /nix/store/j16048gsic45ijdpw82172pfxdz9nw2w-clang-wrapper-16.0.6.drv»
```

Am I doing it wrong?

Comment (Atemu) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:22]: Ah, `makeRustPlatform` doesn't take a `clang` arg but does accept arbitrary args, so it doesn't throw an error to do this.

I don't see another option short of refactoring `makeRustPlatform` then.

I really don't know how this all works together; doesn't rust internally depend on clang aswell? Shouldnt'y you need to override that aswell?

I'd like input of the rust people on this. It appears they were pinged already.

**Comment by diogotcorreia** [comment]
Result of `nixpkgs-review pr 281192` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>postgresql12JitPackages.pgvecto-rs</li>
    <li>postgresql12Packages.pgvecto-rs</li>
    <li>postgresql13JitPackages.pgvecto-rs</li>
    <li>postgresql13Packages.pgvecto-rs</li>
    <li>postgresql14JitPackages.pgvecto-rs</li>
    <li>postgresql14Packages.pgvecto-rs</li>
    <li>postgresqlJitPackages.pgvecto-rs (postgresql15JitPackages.pgvecto-rs)</li>
    <li>postgresqlPackages.pgvecto-rs (postgresql15Packages.pgvecto-rs)</li>
    <li>postgresql16JitPackages.pgvecto-rs</li>
    <li>postgresql16Packages.pgvecto-rs</li>
  </ul>
</details>

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (PhilippWoelfel) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix]: > Relative paths are not part of a package's interface. We'll have to find a different solution.

Sorry, I didn't understand that comment, and I'm curious about this... Why is the relative path a problem? Also, for example, `pkgs/os-specific/darwin/apple-sdk-11.0/default.nix` does something similar:
```
    rustPlatform = pkgs.makeRustPlatform {
      inherit (pkgs.darwin.apple_sdk_11_0) stdenv;
      inherit (pkgs) rustc cargo;
    } // {
      inherit (pkgs.callPackage ../../../build-support/rust/hooks {
        inherit (pkgs.darwin.apple_sdk_11_0) stdenv;
        inherit (pkgs) cargo rustc;
        clang = mkCc pkgs.clang;
      }) bindgenHook;
    };
```

Comment (Atemu) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix]: Relative paths are not guaranteed to remain stable.

People who reorganise one file in nixpkgs should not be expected to check and adjust relative paths in all other files.

This is relevant as we'll soon see the great RFC140 migration.

Now that the hooks override is fixed, the the apple SDK should use this style aswell as it doesn't have this obvious issue. (It may still have less obvious issues.)

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/0001-read-clang-flags-from-environment.diff
Comment (Atemu) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/0001-read-clang-flags-from-environment.diff:13]: Could you add that fact as an explicit comment?

**Review by Atemu — APPROVED** [review]
I don't see any obvious issues on this anymore. I'll wait a few days for input from rust people on how to properly override clang in the rust platform but merge after that.

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/0001-read-clang-flags-from-environment.diff
Comment (diogotcorreia) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/0001-read-clang-flags-from-environment.diff:14]: Done, and squashed/reordered the commits so that it's ready to merged.

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (Mic92) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:22]: This is what we do for firefox: https://github.com/NixOS/nixpkgs/blob/9d422aeafc8780e1efe49f5891b70685d9c05c40/pkgs/applications/networking/browsers/firefox/common.nix#L177

Comment (Mic92) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:22]: Not sure if it also applies to this.

Comment (diogotcorreia) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:22]: @Mic92 Thanks for you help, but it seems that it still doesn't work. Let me know if I'm using this wrong:
```
nix-repl> (pkgs.makeRustPlatform { inherit (pkgs) rustc cargo; stdenv = pkgs.overrideCC pkgs.llvmPackages.stdenv (llvmPackages.stdenv.override { cc = pkgs.clang_17; }); }).bindgenHook.clang
«derivation /nix/store/c9ym6jk1pf0bx62j29g51k00pp16zf9l-clang-wrapper-16.0.6.drv»

nix-repl> (pkgs.makeRustPlatform { inherit (pkgs) rustc cargo; stdenv = pkgs.overrideCC pkgs.llvmPackages.stdenv.override { cc = pkgs.clang_17; }; }).bindgenHook.clang
«derivation /nix/store/c9ym6jk1pf0bx62j29g51k00pp16zf9l-clang-wrapper-16.0.6.drv»

nix-repl> (pkgs.makeRustPlatform { inherit (pkgs) rustc cargo; stdenv = pkgs.llvmPackages.stdenv.override { cc = pkgs.clang_17; }; }).bindgenHook.clang
«derivation /nix/store/c9ym6jk1pf0bx62j29g51k00pp16zf9l-clang-wrapper-16.0.6.drv»
```

For context, this is what I want to achieve (this is the current solution in the code):
```
nix-repl> (pkgs.rustPlatform.bindgenHook.override { clang = pkgs.clang_17; }).clang
«derivation /nix/store/cxxk9njrx1zm5ynmsi0wyy9dv58h2zci-clang-wrapper-17.0.6.drv»
```

(Also, I do want clang 16, but I want to pin it; I'm just using clang 17 in the repl to make sure it is being overridden)

Is there something I'm overlooking?

**Comment by diogotcorreia** [comment]
Final testing remark: I am now using this in my Immich instance and it seems to be running fine :partying_face: 

https://github.com/diogotcorreia/dotfiles/commit/70e71c4a8160a3cf0132a8553b086b29bae749dd

**Review by Atemu — APPROVED** [review]
When a better solution to the rustPlatform clang issue is found, we can always do that later.

Apart from that, this PR has no obvious issues, so I'm inclined to merge.

**Comment by Atemu** [comment]
The aarch64-linux and all Darwin builds appears to be broken. Could you mark them as such?

**Comment by diogotcorreia** [comment]
@Atemu Should I use `meta.badPlatforms` or `meta.broken`?

**Comment by Atemu** [comment]
Just `meta.broken`. I don't think I've ever see `badPlatforms` used for anything and if a build will not produce the expected result, `meta.broken` it is.

**Comment by Atemu** [comment]
https://github.com/ofborg build postgresqlPackages.pgvecto-rs postgresqlPackages.pgvecto-rs.passthru.tests

**Comment by marsam** [comment]
Hi, thanks a lot for your contribution. Sorry for the delay.
Closing in favor of https://github.com/NixOS/nixpkgs/pull/287632 I've integrated your changes there. Thanks!

**Comment by diogotcorreia** [comment]
@marsam 
Hey!
I don't want to come across as rude, but I was surprised to see this PR get closed given that is was ready to merge and been around for longer than the other one (and also the one mentioned in the packaging request issue).
I'm fairly new to contributing to Nixpkgs, so I'm not sure how things work here -- **is there any way to avoid these kinds of collisions in the future?** Before starting a PR, I usually make sure that there is no existing issue/PR of the same thing.

Ultimately, what matters is that this gets added to Nixpkgs and I can't wait for either of the PRs to be merged! :) But it is fairly discouraging to see many of the hours I put into this PR being "thrown out", as you sure understand.

Sorry for the small "rant", and let me know if something is needed on the other PR (it, for example, doesn't seem to pin `cargo-pgrx` like mine does; not sure if something else is missing).

**Comment by marsam** [comment]
Not rude at all; I would word it the same way you did.

I'm so sorry, I should have manage this better, it's totally my fault. It was not nice from me to make this arbitrary decision without consulting you first.

Why did I choose the other PR? somehow, I just had it in my notifications first, and did a quick test. I noticed your PR after; so I thought we could merge your changes in the other PR. It was an arbitrary decision, and I handled it poorly.
In retrospective I should have promoted you PR instead.

I apologize for my behavior and I hope this doesn't discourage you to keep on contributing to nixpkgs.
I'll reflect on how to handle this better in the future.

**Comment by diogotcorreia** [comment]
@marsam Thank you for your reply, I understand that handling so many PRs can be overwhelming :sweat_smile: Thanks for all the work :)
I will for sure keep contributing to nixpkgs :rocket: 

Onto more practical matters: if you think this PR is in a better state than the other one (i.e., it is basically ready to merge), I don't mind updating it to 0.2.1 so that we can get it merged :)
And obviously, I can add @esclear as a maintainer as well, if they so desire.

**Comment by marsam** [comment]
Sure, I think we can reopen this one :)

File: nixos/tests/pgvecto-rs.nix
Comment (marsam) [inline nixos/tests/pgvecto-rs.nix:33]:
```suggestion
      maintainers = [ diogotcorreia ];
```

File: nixos/tests/pgvecto-rs.nix
Comment (Atemu) [reply nixos/tests/pgvecto-rs.nix:69]: This may change in the future, so I think it's fine to keep it.

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (diogotcorreia) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:22]: Just tried this out with `clang_17` and it still built, so it means it doesn't work :(

**Comment by Atemu** [comment]
@ofBorg build postgresqlPackages.pgvecto-rs, postgresqlPackages.pgvecto-rs.passthru.tests

The CI failure is unrelated, you can ignore it.

**Comment by diogotcorreia** [comment]
Just deployed this on my Immich server, everything working perfectly!

**Review by Atemu — APPROVED** [review]
I think this is GTG for an initial implementation.

Could you squash the fixup commits?

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:114]:
```suggestion
    description = "Scalable, Low-latency and Hybrid-enabled Vector Search in Postgres";
```
This should just be a half sentence and not two sentences

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:112]: Can we add a short comment why this isn't working or set platforms to just x86_64-linux?

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:81]:
```suggestion
      "openai_api_rust-0.1.8" = "sha256-os5Y8KIWXJEYEcNzzT57wFPpEXdZ2Uy9W3j5+hJhhR4=";
      "std_detect-0.1.5" = "sha256-RwWejfqyGOaeU9zWM4fbb/hiO1wMpxYPKEjLO0rtRmU=";
```

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:30]:
```suggestion
    pname = "cargo-pgrx";
```

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (diogotcorreia) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:112]: I think @Atemu advocated for `broken` instead of `platforms` here: https://github.com/NixOS/nixpkgs/pull/287632#discussion_r1513851890

As for why it's broken in linux_aarch64, I personally have no idea as I personally don't have any ARM machine to try it out. Judging from [comments in the other PR](https://github.com/NixOS/nixpkgs/pull/287632#issuecomment-1971770586), it seems to be an issue with bindgen that has already been resolved, but pgvecto-rs hasn't incorporated the changes yet; I'll that as a note.
As for darwin, I don't see any official builds from pgvecto-rs, but darwin is listed in the `targets` section of the [`rust-toolchain.toml`](https://github.com/tensorchord/pgvecto.rs/blob/875a20e412f041c2145ce698d6cc545ab0847ec2/rust-toolchain.toml) file upstream, so _in theory_ it should work. Again, I don't own any `darwin` device, so I can't really test this.

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (diogotcorreia) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:53]: Done, thanks for noticing!

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (diogotcorreia) [reply pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:112]: I'll update the comment with the issue link for pgrx since it's more relevant, thanks!

File: pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix
Comment (diogotcorreia) [inline pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix:116]: @esclear Do you want to be added as a maintainer here as well?

**Review by marsam — APPROVED** [review]
LGTM. Thanks a lot!

**Comment by diogotcorreia** [comment]
Just added @esclear as maintainer as well, as requested.

Hopefully that's all :rocket:

**Review by esclear — APPROVED** [review]

**Comment by Atemu** [comment]
Thanks!

**Pull request merged**
